R/period.R

```diff
@@ -308,21 +308,58 @@ CompleteListPeriodNames <- function(x, by)
     observed.dates <- AsDate(x, on.parse.failure = "silent")
     Period(seq(min(observed.dates), max(observed.dates), by = by), by)
 }
+
 #' \code{Period}
 #'
 #' @description Converts a date into a character.
 #' @param x The date.
-#' @param by The period used in the conversion (e.g., "week", "year").
-#' @importFrom lubridate floor_date
+#' @param by The period used in the conversion (e.g., "week", "year"). Special
+#' cases include:
+#' - "quarter", which provides the month for the quarter as YYYY-MM
+#' - "nice.quarter", which provides the quarter in the form Q1 2022
+#' - "2-week", "4-week", etc which provides multi-weekl periods labeled with
+#'   first date in the corresponding period, e.g. "2 weeks commencing 2022-07-03" 
+#'   Requires to use to specify an anchor date as a point of reference.
+#' @param anchor.date The date to use as a reference for multi-week periods. Can be a
+#' Date object, or a string in the form "YYYY-MM-DD". For example,
+#' when wanting two-week periods for "2022-07-04", does this date fall into
+#' the 2 weeks commencing "2022-07-04" or the two weeks comencing "2022-06-27".
+#' @importFrom lubridate floor_date make_difftime
 #' @export
-Period <- function(x, by)
+Period <- function(x, by, anchor.date = "1970-01-01")
 {
+    multi.week = grepl("-week", by, fixed = TRUE)
```

Comment:
If `by` is not provided, then this line will error immediately. This code isn't used in the forthcoming `if` statements until line 344 so it would be better to move it in the line before it. Also, consider using `multi.week <- endsWith(x, "-week")` instead which is more suited.

Reply:
fixed